src/backends/interflop-cancellation/interflop_cancellation.c

```diff
@@ -102,33 +124,41 @@ static inline double _noise_binary64(const int exp) {
        * This particular version in the case of cancellations does not use     \
        * extended quad types */                                                \
       const int32_t e_n = e_z - (cancellation - 1);                            \
-      *Z = *Z + _noise_binary64(e_n);                                          \
+      *Z = *Z + _noise_binary64(e_n, MCA_DATA);                                \
     }                                                                          \
   })
 
 /* Cancellations can only happen during additions and substractions */
 static void _interflop_add_float(float a, float b, float *c,
                                  __attribute__((unused)) void *context) {
   *c = a + b;
-  cancell(a, b, c);
+  _INIT_MCA_DATA(context, mca_data, random_state, random_state_valid,
```

Comment:
This design is strange... Why call _INIT_MCA_DATA here in all these functions. Instead of calling it once from `cancell` ?